experiment/fuzz_target_error.py

```diff
@@ -39,17 +41,24 @@ class SemanticCheckResult:
   SYMPTOM_ASAN = re.compile(r'ERROR: AddressSanitizer: (.*)\n')
   # Matches 'ERROR: libFuzzer: timeout after xxx'
   SYMPTOM_LIBFUZZER = re.compile(r'ERROR: libFuzzer: (.*)\n')
+  # E.g., matches 'SCARINESS: 10 (null-deref)'
+  SYMPTOM_SCARINESS = re.compile(r'SCARINESS:\s*\d+\s*\((.*)\)\n')
 
   NO_COV_INCREASE_MSG_PREFIX = 'No code coverage increasement'
 
   @classmethod
   def extract_symptom(cls, fuzzlog: str) -> str:
     """Extracts crash symptom from fuzzing logs."""
-    match = cls.SYMPTOM_ASAN.match(fuzzlog)
+    # Need to catch this before ASAN.
+    match = cls.SYMPTOM_SCARINESS.search(fuzzlog)
+    if match:
+      return match.group(1)
+
+    match = cls.SYMPTOM_ASAN.search(fuzzlog)
     if match:
       return f'ASAN-{match.group(0)}'
 
-    match = cls.SYMPTOM_LIBFUZZER.match(fuzzlog)
+    match = cls.SYMPTOM_LIBFUZZER.search(fuzzlog)
```

Comment:
@happy-qop: We need to use `search()` instead of [`match()`](https://docs.python.org/3/library/re.html#re.search:~:text=or%20more%20characters-,at%20the%20beginning%20of%20string,-match%20the%20regular) here because the pattern is never the beginning of the `fuzzlog`. This is why our previous report missed the crash reason, e.g., [here](https://llm-exp.oss-fuzz.com/Result-reports/ofg-pr/2024-04-26-228-dg-comparison/sample/output-fribidi-fribidi_log2vis/08).

Now we can correctly capture the crash reason and use the corresponding text in the fixing prompt.
For example, this captured a signal and fixed it:
https://llm-exp.oss-fuzz.com/Result-reports/ofg-pr/2024-04-26-231-dg-comparison/sample/output-libusb-libusb_get_bos_descriptor/06

Reply:
thanks for identifying and fixing this!